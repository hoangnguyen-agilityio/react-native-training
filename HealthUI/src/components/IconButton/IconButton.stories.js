import { storiesOf } from '@storybook/react-native';
import React from 'react';
import IconButton from '.';
import CenterView from '../CenterView';
import ArrowIcon from '../../assets/right-arrow.svg';
import AddIcon from '../../assets/add.svg';

storiesOf('IconButton', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('With next icon', () => (
    <IconButton>
      <ArrowIcon />
    </IconButton>
  ))
  .add('With add icon', () => (
    <IconButton>
      <AddIcon />
    </IconButton>
  ));
