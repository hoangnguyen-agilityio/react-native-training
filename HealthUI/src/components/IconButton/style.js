import colors from "../../constants/colors";

export default {
    button: {
      backgroundColor: colors.bg_purple,
      color: colors.white,
      padding: 16,
      borderRadius: 50,
    }
}