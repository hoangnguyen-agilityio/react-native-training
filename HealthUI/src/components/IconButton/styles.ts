import { StyleSheet } from 'react-native';
import colors from '../../constants/colors';

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.purple,
    color: colors.white,
    padding: 16,
    borderRadius: 50,
  },
});

export default styles;
