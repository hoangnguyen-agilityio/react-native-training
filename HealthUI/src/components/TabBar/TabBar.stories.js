import { storiesOf } from '@storybook/react-native';
import React from 'react';
import TabBar from '.';

storiesOf('TabBar', module)
  .add('default', () => (
    <TabBar />
  ));
