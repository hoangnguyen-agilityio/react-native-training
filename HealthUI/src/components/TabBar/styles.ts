import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
  },
  appBarBg: {
    flex: 1,
    height: 132,
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 44,
  },
  iconGroup: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  btnAdd: {
    flex: 1,
    height: '100%',
    alignItems: 'center'
  }
});

export default styles;
