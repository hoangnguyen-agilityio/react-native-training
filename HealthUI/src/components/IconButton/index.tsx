import React from 'react';
import { TouchableHighlight } from 'react-native';
import style from './style';

const IconButton = (props: any) => {
  return <TouchableHighlight style={style.button} {...props} />;
};

export default IconButton;
