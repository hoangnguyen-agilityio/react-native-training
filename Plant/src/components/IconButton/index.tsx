import React, { FC, ReactNode } from 'react';
import { TouchableOpacity } from 'react-native';
import styles from './styles';
import style from './styles';

interface Props {
  color: string;
  children: ReactNode;
  onPress?: () => void;
  shadow?: boolean;
}

const IconButton: FC<Props> = ({
  color,
  children,
  onPress = () => {},
  shadow = false,
}) => {
  return (
    <TouchableOpacity
      style={[
        style.button,
        shadow ? styles.shadow : null,
        { backgroundColor: color },
      ]}
      onPress={onPress}>
      {children}
    </TouchableOpacity>
  );
};

export default IconButton;
