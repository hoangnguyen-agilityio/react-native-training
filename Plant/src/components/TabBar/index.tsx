import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import React, { FC, useContext } from 'react';
import { ImageBackground, Text, TouchableOpacity, View } from 'react-native';
import HomeIcon from '../../assets/images/home.svg';
import ProfileIcon from '../../assets/images/profile.svg';
import AddIcon from '../../assets/images/add.svg';
import COLORS from '../../constants/colors';
import { NavigationListType } from '../../constants/navigationList';
import IconButton from '../IconButton';
import styles from './styles';
import { BottomTabBarProps } from '@react-navigation/bottom-tabs';

const TabBar: FC<BottomTabBarProps> = ({ state, navigation }) => {
  const { routes } = state;

  const onPress = (screen: keyof NavigationListType, index: number) => {
    const route = routes?.find(route => route.name === screen);
    const event = navigation.emit({
      type: 'tabPress',
      target: route?.key,
      canPreventDefault: true,
    });

    if (index !== state.index && !event.defaultPrevented && route?.name) {
      navigation.navigate({ name: route?.name, params: { merge: true } });
    }
  };

  const getColor = (index: number): string =>
    index === state.index ? COLORS.GREEN : COLORS.GRAY;

  return (
    <ImageBackground
      source={require('../../assets/images/tab-bar.png')}
      style={styles.root}
      resizeMode="cover">
      <View style={styles.content}>
        <View style={styles.col}>
          <TouchableOpacity
            style={styles.center}
            onPress={() => onPress('Home', 0)}>
            <HomeIcon width={25} height={25} color={getColor(0)} />
            <Text style={[styles.label, { color: getColor(0) }]}>HOME</Text>
          </TouchableOpacity>
        </View>
        <View style={[styles.col, { marginBottom: 36 }]}>
          <IconButton color="#48A2F5">
            <AddIcon />
          </IconButton>
        </View>
        <View style={styles.col}>
          <TouchableOpacity
            style={styles.center}
            onPress={() => onPress('Profile', 1)}>
            <ProfileIcon width={25} height={25} color={getColor(1)} />
            <Text style={[styles.label, { color: getColor(1) }]}>PROFILE</Text>
          </TouchableOpacity>
        </View>
      </View>
    </ImageBackground>
  );
};

export default TabBar;
