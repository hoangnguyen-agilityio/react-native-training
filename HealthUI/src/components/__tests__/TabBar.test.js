import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from '../IconButton';
import TabBar from '../TabBar';

test('renders correctly', () => {
  const tree = renderer.create(<TabBar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('component elements', () => {
  const tree = renderer.create(<TabBar />).root;

  expect(tree.findAllByType(IconButton)).toHaveLength(1);
});
