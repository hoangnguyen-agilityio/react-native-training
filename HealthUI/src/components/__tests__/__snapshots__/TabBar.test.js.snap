// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 132,
        "paddingHorizontal": 44,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/tab-bar.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 132,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <SvgMock />
      <SvgMock />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5142AB",
            "borderRadius": 50,
            "color": "#FFFFFF",
            "padding": 16,
          }
        }
      >
        <SvgMock />
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <SvgMock />
      <SvgMock />
    </View>
  </View>
</View>
`;
